### PR TITLE
http auth support

### PR DESCRIPTION
**Pull request by acvetkov**
@vitalets 
- append httpAuth support
- fix tests

https://github.com/acvetkov/teamcity-client/issues/3

### CODE REVIEW

File: src/artifact/url.js [hunk 1]
```diff
@@ -3,56 +3,47 @@
...
 /**
- * @param {TeamcityApiOptions} options
  * @param {Object} locatorObject
  * @param {String} path
  * @returns {String}
  */
-export function contentUrl(options, locatorObject, path) {
-    const baseUrl = apiUrl(options);
+export function contentUrl(locatorObject, path) {
     const locator = locatorToString(locatorObject);
```
Comment (vitalets) [inline src/artifact/url.js:14]: maybe add `.toString()` method to locator object?
Then you will no need to import locatorToString and call it many times, just `..${locator}..`

File: src/utils/http-client.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
+/**
+ * @author acvetkov@yandex-team.ru
```
Comment (vitalets) [inline src/utils/http-client.js]: Describe what this client can do?

**Comment by vitalets** [comment]
:+1:

File: src/artifact/url.js (hunk 1 above)
Comment (acvetkov) [inline src/artifact/url.js:14]: `locatorObject` is pure `js` object, that specified by client.
For example:

``` js
var locator = {
  buildType: {
    id: 123
  }
};

client.build.detail(locator)
  .then(processResponse);
```

I think, it's harmfull to force user import additional class from `teamcity-client` package.

``` js
var client = require('teamcity-client');
var Locator = require('teamcity-client/locator');

var locator = new Locator({
  buildType: {
    id: 123
  }
});
client.build.detail(locator)
  .then(processResponse);
```

File: src/utils/http-client.js (hunk 2 above)
Comment (acvetkov) [inline src/utils/http-client.js]: ok

**Pull request merged**